File: backend _Spring Boot/backend/src/main/java/com/eventosperu/backend/model/Servicio.java
package com.eventosperu.backend.model;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "servicios")
public class Servicio {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idServicio;

    @ManyToOne
    @JoinColumn(name = "id_proveedor", nullable = false)
    private Proveedor proveedor;

    @ManyToOne
    @JoinColumn(name = "id_evento", nullable = false)
    private Evento evento;

    @Column(name = "nombre_servicio", nullable = false, length = 100)
    private String nombreServicio;

    @Column(columnDefinition = "TEXT")
    private String descripcion;

    @Column(nullable = false)
    private Double precio;

    @Column(name = "url_foto", length = 255)
    private String urlFoto;
}

File: frontend_Angular/eventos_peru/src/app/modelos/servicio.ts
export interface ServicioRelacionado {
  id?: number;
  nombre?: string;
}

export interface Servicio {
  idServicio: number;
  nombreServicio: string;
  descripcion?: string;
  precio: number;
  proveedor?: ServicioRelacionado & { nombreEmpresa?: string; contacto?: string };
  evento?: ServicioRelacionado & { nombreEvento?: string; tipo?: string };
}
